test/e2e_node/resource_metrics_test.go

```diff
@@ -73,7 +73,7 @@ var _ = SIGDescribe("ResourceMetricsAPI [NodeFeature:ResourceMetrics]", func() {
 			memoryCapacity := node.Status.Capacity["memory"]
 			memoryLimit := memoryCapacity.Value()
 
-			matchResourceMetrics := gstruct.MatchAllKeys(gstruct.Keys{
+			matchResourceMetrics := gomega.And(gstruct.MatchKeys(gstruct.IgnoreMissing, gstruct.Keys{
```

Comment:
how hard is to detect if the swap is neabled in the node programmatically, and just use different `MatchAllKeys` in the two flows?

Reply:
I guess we can fix the failure and iterate later

Reply:
I don't know of an easy way to do so.

Even is `NodeSwap` feature-gate is enabled, it doesn't mean that the node actually has swap provisioned on it. We could always try to find the node that the pod runs on and try executing stuff on it to find this out, but I think it's a big enough complication.

Reply:
> I guess we can fix the failure and iterate later

Sounds good to me :+1: Thanks!

Reply:
thinking aloud, maybe we can use `swapon`? Anyway, the more I think about it the more it seems wiser to fix the burning issue and improve later, the current approach seems good enough for the time being.

If you resubmit for whatever reason, please add copy this content https://github.com/kubernetes/kubernetes/pull/119421#issuecomment-1641892456 as code comment right above the `gomega.And`